### ISSUE TITLE
Questions on benchmark

### ISSUE THREAD

**Issue by JrPeng** [ShuLiu1993/PANet#5]
I find that you use 'roi_Xconv1fc_gn_head_panet' training R-50-PANet and get a 39.8 box mAP. I tried to train a panet using '2fc_mlp_head' with PA-structure unchanged and got a 37.1 box mAP, which is only 0.4 above the fpn baseline. Is there some wrong with this implementation, or is there any probability that PA-net has to be combined with a heavy head?

**Comment by ShuLiu1993** [comment]
Hi,

I have uploaded the model trained with 2fc. The performance should be 39.6 box AP. Please try with the new model and corresponding config file.

Thanks!

**Comment by JiamingSuen** [comment]
Hi @ShuLiu1993 , thanks for this wonderful work! I wonder if pretrained model on Cityscapes can be provided(or converted) in this PyTorch implementation? Thanks!

**Comment by JrPeng** [comment]
Thank you for your response, I will try with your advice. : )

**Comment by JrPeng** [comment]
What if you use only '2fc_mlp_head' rather than '2fc_mlp_with_gn_head', which is more precise on evaluating the influence of 'pa structure'?

**Comment by JrPeng** [comment]
I tried to train the panet on res50-fpn without using any gn to get a precise evaluation on the influence of PA-structure and get the  37.1 bbox mAP. I find that your new model still use GN in both fpn-part and the 2fc-head. Does it mean that GN is really important for training PA-structure?

**Comment by ShuLiu1993** [comment]
@JiamingSuen 
Thanks a lot for your interest! Sorry that I really don't have much time on this currently. Maybe you can try this by yourself since the pre-trained model on COCO is already released.

**Comment by ShuLiu1993** [comment]
@JrPeng 
In our origin implementation, we used Sync BN. In this version, we us GN instead. Using this kind of normalization can help the network converge better and achieve better performance.

**Comment by JiamingSuen** [comment]
@ShuLiu1993 I can totally understand, thanks for your reply!

**Comment by JrPeng** [comment]
Thanks for your reply. I will try to figure out recently. Thank you.

**Comment by JiamingSuen** [comment]
@ShuLiu1993 I'm trying to reproduce cityscapes training with COCO pretrained model, and I'm currently having 34.4 mask AP on Cityscapes. I used the finetuing schedule described in the Mask-RCNN paper(4k iter in which reduce lr at 3k with initial lr 0.01 and 8 batch size). Did you choose the same finetuning steps and lr schedule in PANet? Thanks very much.

**Comment by JiamingSuen** [comment]
@ShuLiu1993 Any comment please? I didn't find any reference about finetuning parameters in your paper. Thanks very much!

**Comment by ShuLiu1993** [comment]
@JiamingSuen Sorry for the late reply. We have listed several hyper-parameters in our paper, which are borrowed from Mask R-CNN. In other words, we strictly followed the training parameters used by Mask R-CNN.

**Comment by Z0org** [comment]
@JiamingSuen can you give me a short tutorial on how to finetune the pretrained coco model? I would like to finetune the model for pedestrian detection on the Kitti Dataset but I don't know where to start.

**Comment by JiamingSuen** [comment]
You may start by trying to adapt cityscapes loader/model to KITTI since KITTI segmentation dataset is using cityscapes format. Cityscapes tools In this codebase would be helpful.

**Comment by JrPeng** [comment]
@Panxj I find that baseline of this implementation is higher than baseline of Detectron.pytorch by Roy. For example, res50-FPN without GN in this implementation has a 38.3box AP while it has a 37.7box AP in Roy's implementation. Besides, I add only PA structure in Roy's code, and find it only contributes 0.3AP improvment, but GN/adaptive fusion contributes a lot. Maybe there is something wrong with my implementation. You can try yourself and we can discuss based on your results.

**Comment by thangvubk** [comment]
@ShuLiu1993 thank you for sharing your work. I had run the ablation on PAnet with ``panet_R-50-FPN_1x_det_2fc.yaml`` config file. I also found that this repo has better baseline compared to original pytorch repo or official detectron in caffe. Without any GN, buttom up path, ada feature pool and multi scale training, this repo can achieve 37.9 mAP, it is higher than official detectron by 1.2 mAP. It seems like your original panet version using sync BN has different improvement factors. But could you please clear us why here we get so good baseline :D. 

| GN | BU path | ada pool | ms train | Result |                                                         Note                                                        |
|:--:|:-------:|:--------:|:--------:|:------:|:-------------------------------------------------------------------------------------------------------------------:|
|  x |    x    |     x    |     x    |  39.6  | Default panet_R-50-FPN_1x_det_2fc.yaml                                                                              |
|  x |    x    |     x    |          |   39   | <details><summary>config</summary>`SCALES=(1000,)`</details>                                                                                                  |
|  x |    x    |          |          |  38.5  | <details><summary>config</summary>`SCALES=(1000,)`<br/> `ROI_BOX_HEAD=fast_rcnn_heads.roi_2mlp_head_gn`</details>                                          |
|  x |         |          |          |  38.4  | <details><summary>config</summary>`SCALES=(1000,)`<br/> `ROI_BOX_HEAD=fast_rcnn_heads.roi_2mlp_head_gn`<br/>`CONV_BODY:FPN.fpn_ResNet50_conv5_body`</details>   |
|    |         |          |          |  37.9  | <details><summary>config</summary>`SCALES=(1000,)`<br/> `ROI_BOX_HEAD=fast_rcnn_heads.roi_2mlp_head_gn`<br/>`CONV_BODY:FPN.fpn_ResNet50_conv5_body`<br/>`FPN.USE_GN=False`</details>  |

**Comment by ShuLiu1993** [comment]
Hi guys, thanks a lot for your interests!

This codebase is heavily based on Detectron.pytorch by Roy. In this codebase and released configs, I used multi-scale training larger testing scale as I noted in the paper. This may be the main reason that the baseline is with a better performance. I also made some minor modifications may also contribute a little bit. 

With GN or SyncBN do help PANet to achieve a better performance with other settings kept the same. They can help the network to converge better. So that's why we should try GN or SyncBN first. Currently I haven't compared GN and SyncBN under the same codebase. But I think SyncBN will achieve comparable or even better performance.

**Comment by zimenglan-sysu-512** [comment]
hi @thangvubk 
i use bu-path, ada-pool, and ms-train, and test use `MIN_SIZE_TEST: 1000`, only get 38.1 mAP. is there something i miss?

**Comment by thangvubk** [comment]
@zimenglan-sysu-512 just use ``panet_R-50-FPN_1x_det_2fc.yaml`` config file and dont make any modifications.

**Comment by zimenglan-sysu-512** [comment]
hi @thangvubk 
do u try soft-nms instead of standard-nms? if use, can u share the results here?

**Comment by thangvubk** [comment]
@zimenglan-sysu-512  For bu_path + ada-pool + ms, i didnt make any modification, just clone the network and train. Did you do the same thing?

**Comment by zimenglan-sysu-512** [comment]
i did not use this repo, i just add some code `maskrcnn-benchmark` (w/o GN).

**Comment by thangvubk** [comment]
@zimenglan-sysu-512 It is hard to say when you implement in other repo. Ussually, it doesnt work as expected due to underlining implementation details. You can try to add GN to see if their is any improvement.

**Comment by zimenglan-sysu-512** [comment]
thanks @thangvubk 
i will try to add GN to FPN and heads.

**Comment by zimenglan-sysu-512** [comment]
hi @thangvubk 
i add GN to FPN and RoI head, but the performance only gets 38.3%. btw, when training, since the gpu memory limits, so each gph holds 1 image.

**Comment by thangvubk** [comment]
@zimenglan-sysu-512  i'm not sure where u are wrong. Btw, mmdet is planing to release PANet also, see [here](https://github.com/open-mmlab/mmdetection/issues/96).

**Comment by zimenglan-sysu-512** [comment]
@thangvubk 
i set two images per gpu, the performance can get 0.388, which is still a little gap comparing with the paper.

**Comment by zimenglan-sysu-512** [comment]
hi @thangvubk 
i guess the performance that drops a little is due to that i use `2fc` instead of `Xconv1fc`.

**Comment by LaoYang1994** [comment]
@zimenglan-sysu-512 I also re-implement the panet using maskrcnn_benchmark（2mlp without ms train， gn， sbn）. But the performance is worse than this repo. If convenient, can you share your code with me? I just want to make sure whether there is something wrong with my code.
